### PR TITLE
feat: Add makefile

### PR DESCRIPTION
**Pull request by pythonbrad**
Linked to the issue #7

### CODE REVIEW

**Review by Sanix-Darker — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+##install-deps: setup your dev environment
+install-deps: venv $(CONFIG_FILE)
+	$(PIP) install --upgrade pip
+	$(PIP) install -r requirements.txt
+	$(PIP) install flake8
+	$(PIP) install --upgrade build
+
+##run: run the api locally
+run: install-deps
+	$(PYTHON) broken_link_checker $(link) --delay 1
```
Comment (Sanix-Darker) [inline Makefile:26]: This should be run with an `-m`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+lint: install-deps venv
+	$(PYTHON) -m flake8 src --show-source --statistics
+
+build: install-deps
+	$(PYTHON) -m build
+
+##test: test your code
+test: build
+	$(PYTHON) -m unittest
+	$(PIP) install dist/blc-*.whl
```
Comment (Sanix-Darker) [inline Makefile:37]: why test with the wheel ?

**Review by Sanix-Darker — CHANGES_REQUESTED** [review]

**Comment by elhmn** [comment]
The Makefile, was not running the application correctly. I did some updates and it should be working now. 

The only issue is that the test suite should be updated as it is now broken, and I would recommend to update the README in other to use Makefiles target to run the application instead of raw python scripts

File: Makefile (hunk 2 above)
Comment (pythonbrad) [reply Makefile:34]: It is because i want test:
- the build of the package
- the installation of the package
- the code

**Pull request merged**